main/src/ca/uwaterloo/flix/language/phase/Simplifier.scala

```diff
@@ -50,285 +50,289 @@ object Simplifier {
     }
 
     /**
-      * Translates the given expression `exp0` to the SimplifiedAst.
-      */
-    def visitExp(exp0: TypedAst.Expression): SimplifiedAst.Expression = exp0 match {
-      case TypedAst.Expression.Var(sym, tpe, loc) => SimplifiedAst.Expression.Var(sym, tpe, loc)
-
-      case TypedAst.Expression.Def(sym, tpe, loc) => SimplifiedAst.Expression.Def(sym, tpe, loc)
-
-      case TypedAst.Expression.Sig(sym, tpe, loc) => SimplifiedAst.Expression.HoleError(new Symbol.HoleSym(sym.clazz.namespace, sym.name, loc), tpe, loc) // TODO replace with implementation
-
-      case TypedAst.Expression.Hole(sym, tpe, loc) => SimplifiedAst.Expression.HoleError(sym, tpe, loc)
-
-      case TypedAst.Expression.Unit(loc) => SimplifiedAst.Expression.Unit(loc)
-
-      case TypedAst.Expression.Null(tpe, loc) => SimplifiedAst.Expression.Null(tpe, loc)
-
-      case TypedAst.Expression.True(loc) => SimplifiedAst.Expression.True(loc)
-
-      case TypedAst.Expression.False(loc) => SimplifiedAst.Expression.False(loc)
-
-      case TypedAst.Expression.Char(lit, loc) => SimplifiedAst.Expression.Char(lit, loc)
-
-      case TypedAst.Expression.Float32(lit, loc) => SimplifiedAst.Expression.Float32(lit, loc)
-
-      case TypedAst.Expression.Float64(lit, loc) => SimplifiedAst.Expression.Float64(lit, loc)
-
-      case TypedAst.Expression.Int8(lit, loc) => SimplifiedAst.Expression.Int8(lit, loc)
-
-      case TypedAst.Expression.Int16(lit, loc) => SimplifiedAst.Expression.Int16(lit, loc)
-
-      case TypedAst.Expression.Int32(lit, loc) => SimplifiedAst.Expression.Int32(lit, loc)
-
-      case TypedAst.Expression.Int64(lit, loc) => SimplifiedAst.Expression.Int64(lit, loc)
-
-      case TypedAst.Expression.BigInt(lit, loc) => SimplifiedAst.Expression.BigInt(lit, loc)
+     * Translates the given expression `exp0` to the SimplifiedAst.
+     */
+    def visitExp(exp0: TypedAst.Expression): SimplifiedAst.Expression = {
+      val purity = isPure(exp0.eff)
```

Comment:
Please don't do this. Its bad form, just do it locally.

Reply:
Because visitExp is so long, if you do this purity is essentially a global variable, which we dont want.